db-engine/src/main/java/com/splicemachine/db/impl/sql/GenericPreparedStatement.java

```diff
@@ -367,11 +367,13 @@ public ResultSet executeStmt(Activation activation, boolean rollbackParentContex
 
             ResultSet resultSet = null;
             try {
-
                 resultSet = activation.execute();
 
                 resultSet.open();
             } catch (StandardException se) {
+                if (resultSet != null) {
+                    resultSet.close();
```

Comment:
This should be closed in a finally block too, or use try with resources.  If a RuntimeException is thrown we might miss closing the result set.  There are a bunch of other places in the code where ResultSets are opened and not closed when an exception is thrown.  Should we handle them through this PR?  If not, I can open another Jira.

Reply:
I think we can't use a try-with-resources because we are returning the resultSet, but I think we should be broader on the type of Exception caught, and also extend the try{} block until we return the resultSet, so that we close it if we raise an exception before exiting.

Reply:
Done 👍

Reply:
> There are a bunch of other places in the code where ResultSets are opened and not closed when an exception is thrown. Should we handle them through this PR? If not, I can open another Jira.
I created DB-12131 for that